src/App.vue

```diff
@@ -732,15 +727,15 @@ export default class App extends Mixins(
     // add this as a filing item
     const filingItem = {
       availableOnPaperOnly: header.availableOnPaperOnly,
-      businessIdentifier: this.getIdentifier,
-      commentsCount: application.commentsCount,
-      commentsLink: application.commentsLink,
-      displayLedger: application.displayLedger,
+      businessIdentifier: filing.business.identifier || this.getIdentifier,
```

Comment:
This is key -- save the new business id (eg, BCxxx) if it's present (which it is if the filing is completed), else fall back to the current idenfier (eg, Txxx).